documentation/content/main/start-here/username-password.nuxt.md

```diff
@@ -363,3 +365,60 @@ export default defineEventHandler(async (event) => {
 	return null;
 });
 ```
+
+### Handle authentication with state and middleware
+
+Using middlewares and composables is a really neat way to avoid duplicating your authorization logic.
+
+Create a `useAuth` composable :
+
+```ts
+// composables/useAuth.ts
+import type { User } from "lucia-auth";
+
+export const useAuth = () => {
+	const user = useState<User | null>("user", () => null);
+
+	const fetchUser = async () => {
+		const { data } = await useFetch<{ user: User }>("/api/user");
```

Comment:
We shouldn't need to type the fetch request here - is the type inference not working from `/api/user`?

Reply:
Somehow it wasn't worked with pnpm hoisting disabled